phoenix-core/src/main/java/org/apache/phoenix/index/IndexMaintainer.java

```diff
@@ -612,6 +612,10 @@ private Void addDataColInfo(final PTable dataTable, Expression expression) {
                         byte[] cf = colExpression.getColumnFamily();
                         byte[] cq = colExpression.getColumnQualifier();
                         try {
+                            if (dataTable.hasOnlyPkColumns()) {
```

Comment:
@virajjasani can we move this into the catch block as @stoty was also suggesting ?

Reply:
yeah now it seems doable, meaning it is clean even when we put in catch.
done

Reply:
Thanks @virajjasani .